lib/nsq/connection.rb

```diff
@@ -32,6 +32,7 @@ def initialize(opts = {})
       @msg_timeout = opts[:msg_timeout] || 60_000 # 60s
       @max_in_flight = opts[:max_in_flight] || 1
       @ssl_context = opts[:ssl_context]
+      @tls_v1 = !!opts[:tls_v1]
       validate_ssl_context! if @ssl_context
```

Comment:
If you provide:
` { tls_v1: false, ssl_context: { key: _, cert: _ } }` TLS will not be used but the ssl_context will still be validated.
Given the ssl_context does not validate, errors will still be raised which may be confusing when you have `tls_v1: false`